lib/omniauth/auth0/jwt_validator.rb

```diff
@@ -7,17 +7,22 @@ module OmniAuth
   module Auth0
     # JWT Validator class
     class JWTValidator
-      attr_accessor :issuer
+      attr_accessor :issuer, :domain
 
       # Initializer
       # @param options object
       #   options.domain - Application domain.
+      #   options.issuer - Application issuer (optional).
       #   options.client_id - Application Client ID.
       #   options.client_secret - Application Client Secret.
       def initialize(options)
-        temp_domain = URI(options.domain)
-        temp_domain = URI("https://#{options.domain}") unless temp_domain.scheme
-        @issuer = "#{temp_domain}/"
+        # Use custom issuer if provided, otherwise use domain
+        @issuer = if options.respond_to?(:issuer)
+                    uri_string(options.issuer)
+                  else
+                    uri_string(options.domain)
+                  end
+        @domain = uri_string(options.domain)
```

Comment:
this line can be moved above `@issuer` and you can use this value for the "else" case and avoid calling `uri_string` again.

Reply:
```ruby
@domain = uri_string(options.domain)
@issuer = @domain
@issuer = uri_string(options.issuer) if options.respond_to?(:issuer)
```